src/test/java/jUnitTesting/ShortestPathCalculatorTest.java

```diff
@@ -0,0 +1,85 @@
+package jUnitTesting;
+
+import org.example.ShortestPathCalculator;
+import org.example.model.Point;
+import org.example.model.Route;
+import org.example.model.builder.PointBuilder;
+import org.example.model.builder.RouteBuilder;
+import org.junit.Before;
+import org.junit.Test;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class ShortestPathCalculatorTest {
+    private List<Point> points;
+    private List<Route> routes;
+    private ShortestPathCalculator calculator;
+
+    @Before
+    public void setUp() {
+        // Creating some test data
+        Point pointA = new PointBuilder().setCityName("Miami").getPoint();
+        Point pointB = new PointBuilder().setCityName("Chicago").getPoint();
+        Point pointC = new PointBuilder().setCityName("Algonquin").getPoint();
+        Point pointD = new PointBuilder().setCityName("Washington").getPoint();
+
+        points = Arrays.asList(pointA, pointB, pointC, pointD);
+
+        Route routeAB = new RouteBuilder().setStartPoint(pointA).setEndPoint(pointB).setDistance(10).getRoute();
+        Route routeAC = new RouteBuilder().setStartPoint(pointA).setEndPoint(pointC).setDistance(15).getRoute();
+        Route routeBC = new RouteBuilder().setStartPoint(pointB).setEndPoint(pointC).setDistance(5).getRoute();
+        Route routeCD = new RouteBuilder().setStartPoint(pointC).setEndPoint(pointD).setDistance(20).getRoute();
+
+        routes = Arrays.asList(routeAB, routeAC, routeBC, routeCD);
+
+        calculator = new ShortestPathCalculator(points, routes);
+    }
+
+    @Test
+    public void testCalculateShortestPath() {
+        // Testing shortest path from A to D
+        long shortestPathAD = calculator.calculateShortestPath(points.get(0), points.get(3));
+        assertEquals(25, shortestPathAD);
```

Comment:
Shouldn't this be 35 instead of 25?
AC + CD = AD, right?
Therefore, 15 + 20 = 35.

Reply:
i agree